components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -378,11 +383,11 @@ func (o *odometry) trackPosition(ctx context.Context) {
 			o.orientation.Yaw = math.Mod(o.orientation.Yaw+oneTurn, oneTurn)
 
 			// Calculate X and Y by using centerDist and the current orientation yaw (theta).
-			o.position.X += (centerDist * math.Cos(o.orientation.Yaw))
-			o.position.Y += (centerDist * math.Sin(o.orientation.Yaw))
+			o.position.Y += (centerDist * math.Cos(o.orientation.Yaw))
+			o.position.X += (centerDist * math.Sin(o.orientation.Yaw))
 
-			distance := math.Hypot(o.position.X, o.position.Y)
-			heading := math.Atan2(o.position.Y, o.position.X)
+			distance := math.Hypot(o.position.Y, o.position.X)
```

Comment:
This works, but a future reader will wonder why we defined and used Y before X. Mild preference for swapping lines 386 and 387, and putting line 389 back the way it originally was. That way, we define and use X before Y like usual.